### ISSUE TITLE
InventoryCloseEvent is called twice on AnvilGUIBuilder::open()

### ISSUE THREAD

**Issue by woodyn1002** [WesJD/AnvilGUI#339]
```kotlin
@EventHandler
fun onInventoryClick(event: InventoryClickEvent) {
    if (!isCustomGui(event.inventory)) return

    if (event.rawSlot == 0) {
        val anvilGuiBuilder = createAnvilGuiBuilder()
        anvilGuiBuilder.open(event.whoClicked as Player)
    }
}

@EventHandler
fun onInventoryClose(event: InventoryCloseEvent) {
    println(event)
}
```
```
[19:09:36] [Server thread/INFO]: org.bukkit.event.inventory.InventoryCloseEvent@6d7d67f
[19:09:37] [Server thread/INFO]: org.bukkit.event.inventory.InventoryCloseEvent@262b0318
```

While the player has an inventory opened, `AnvilGUIBuilder::open()` calls `InventoryCloseEvent` twice.
tested in 1.20.4

**Comment by KRIPA-YT** [comment]
Confirm, is the same on 1.21 aswell, it really messes up my plugin logic

**Comment by 0dinD** [comment]
@woodyn1002 @KRIPA-YT I'm quite certain that I've found the problem and a fix for it, see #346. Feel free to build and use AnvilGUI from my PR branch if you want to help verify that it works.
